test/functional/test_framework/test_framework.py

```diff
@@ -431,10 +431,11 @@ def _initialize_chain(self):
             self.nodes = []
             self.disable_mocktime()
             for i in range(MAX_NODES):
-                os.remove(log_filename(self.options.cachedir, i, "debug.log"))
-                os.remove(log_filename(self.options.cachedir, i, "wallets/db.log"))
-                os.remove(log_filename(self.options.cachedir, i, "peers.dat"))
-                os.remove(log_filename(self.options.cachedir, i, "fee_estimates.dat"))
+                os.remove(os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", "debug.log"))
+                os.remove(os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", "mempool.dat"))
+                os.remove(os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", "wallets/db.log"))
+                os.remove(os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", "peers.dat"))
+                os.remove(os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", "fee_estimates.dat"))
```

Comment:
Would be nice to avoid so much duplication, e.g. 
```python
def rm_cache_path(filename):
    os.remove(os.path.join(get_datadir_path(self.options.cachedir, i), "regtest", filename))
```
but this is fine as-is.